src/main/scala/is/hail/table/Table.scala

```diff
@@ -354,70 +354,24 @@ class Table(val hc: HailContext, val tir: TableIR) {
     annotateGlobal(ann, t, name)
   }
 
-  def annotateGlobalExpr(expr: String): Table = {
+  def selectGlobal(expr: String): Table = {
     val ec = EvalContext("global" -> globalSignature)
     ec.set(0, globals.value)
 
-    val (paths, types, f) = Parser.parseAnnotationExprs(expr, ec, None)
-
-    val inserterBuilder = new ArrayBuilder[Inserter]()
-
-    val finalType = (paths, types).zipped.foldLeft(globalSignature) { case (v, (ids, signature)) =>
-      val (s, i) = v.insert(signature, ids)
-      inserterBuilder += i
-      s.asInstanceOf[TStruct]
-    }
-
-    val inserters = inserterBuilder.result()
-
-    val ga = inserters
-      .zip(f())
-      .foldLeft(globals.value) { case (a, (ins, res)) =>
-        ins(a, res).asInstanceOf[Row]
-      }
-
-    copy2(globals = globals.copy(value = ga, t = finalType),
-      globalSignature = finalType)
-  }
-
-  def selectGlobal(fields: java.util.ArrayList[String]): Table = {
-    selectGlobal(fields.asScala.toArray: _*)
-  }
-
-  def selectGlobal(fields: String*): Table = {
-    val ec = EvalContext("global" -> globalSignature)
-    ec.set(0, globals.value)
-
-    val (paths, types, f) = Parser.parseSelectExprs(fields.toArray, ec)
-
-    val names = paths.map {
-      case Left(n) => n
-      case Right(l) => l.last
-    }
-
-    val overlappingPaths = names.counter().filter { case (n, i) => i != 1 }.keys
-
-    if (overlappingPaths.nonEmpty)
-      fatal(s"Found ${ overlappingPaths.size } ${ plural(overlappingPaths.size, "selected field name") } that are duplicated.\n" +
-        "Overlapping fields:\n  " +
-        s"@1", overlappingPaths.truncatable("\n  "))
+    val ast = Parser.parseToAST(expr, ec)
+    assert(ast.`type`.isInstanceOf[TStruct])
 
-    val inserterBuilder = new ArrayBuilder[Inserter]()
+    ast.toIR() match {
+      case Some(ir) =>
```

Comment:
You need to check there are no aggregators.  I assume you're planning to do that as a second step?

Reply:
In particular, @rcownie had the same problem and I suggested he write a IR predicate to check for aggregators, see: https://github.com/hail-is/hail/pull/3174#discussion_r175643610.  One of you should probably PR that as a standalone change so you can both use it.  Can you coordinate with him?

Reply:
That was next on my list so I'll split it off and try to do it quickly

Reply:
The eval context for annotate/select globals only has "global" in it. I don't think you can do an aggregation.

Reply:
Great,  I stand correct.  @rcownie if you could still do ContainsAggregation next, that'd be great, because even if this PR doesn't need it, @jigold's next one will (aggregations directly).